pipenv/environments.py

```diff
@@ -276,6 +276,12 @@ def is_quiet(threshold=-1):
     return PIPENV_VERBOSITY <= threshold
 
 
+def is_in_virtualenv():
```

Comment:
You don't like using globals everywhere to figure this out?

Reply:
Yes the global environment variable is good and convenient but I need a way to get the right status when the environment is modified in runtime.(see the test case)

It isn't a good idea to change the logic only for testing though, because in real case the environment is always decided before any command is launched.

If you are not for this, I can revert these bits and consider using patching in test case.